person.rb

```diff
@@ -25,4 +26,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(book, date)
+    @rentals.push(Rental.new(date, book, self))
+  end
```

Comment:
- [ ] There's no need to push this rental instance here as instantiating a Rental object already does that, remember. This code would mean double addition of the same rental. We don't really need that. Just as the previous reviewer suggested keep the other implementation as it was just re-arrange your arguments:

  ```
    def add_rental(book, date)
      Rental.new(date, book, self)
  end
  ```